### PR TITLE
Add NPM Module for Windows Registry

### PR DESCRIPTION
**Pull request by ritazh**
- [x] Provide link to NPM module
- [x] Provide link to GitHub repo
- [x] Code snippet to demonstrate usage
- [x] Update README to include new section

Ref #37

### CODE REVIEW

**Comment by mousetraps** [comment]
Wow, this is awesome! Right now I'm wondering where this should live, though... @felixrieseberg thoughts? Do we want a section for "walkthroughs" that we can link to from the main content?

**Comment by felixrieseberg** [comment]
@ritazh: Super well done, thanks :clap:! I agree with @mousetraps that should probably have some kind of structure to ensure that we don't put all walkthroughs into one markdown file.

What do you think about this: We put small tips (like `require('os').EOL`) into the cross-platform file - and walkthroughs like the registry one in either a folder (or just into root with a prefix), linking to them. I wanted to add a bunch of stuff around Christmas and I think that'd be a neat way for people to learn about how to do various Windows-things from Node.

**Comment by SteveALee** [comment]
:clap: indeed!

**Comment by ritazh** [comment]
@mousetraps @felixrieseberg I'm happy to make whatever change necessary to get the PR accepted. Let me know!

**Comment by ritazh** [comment]
@mousetraps @felixrieseberg Any updates or feedback on this so we can get this merged?

**Comment by ritazh** [comment]
@mousetraps Anything I can do to get this merged? :smile:

**Comment by Akamaozu** [comment]
Particularly useful information if you're using electron. Well done, @ritazh

**Comment by ritazh** [comment]
Thanks @Akamaozu!

**Comment by jasonyueyang** [comment]
I'm fine to get the content in and refactor later if needed.

**Comment by mousetraps** [comment]
Is there any reason to use windows-registry over winreg, which doesn't require the use of native addons?
https://www.npmjs.com/package/winreg

**Comment by felixrieseberg** [comment]
Yep - @ritazh and @seoudard built it during a hackathon with GitHub/Electron, since just calling regedit.exe and hoping for the best has a bunch of limitations and is tremendously error-prone. The best example that comes to mind is setting file associations and similar tasks requiring self-elevation.

File: README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Note that this is not intended to be a comprehensive set of recommendations. Rat
   * [MAX_PATH explanation and workarounds](windows-environment.md#max_path-explanation-and-workarounds)
   * [Compiling native Addon modules](windows-environment.md#compiling-native-addon-modules)
 * [Building for cross-platform](building-for-cross-platform.md)
+  * [NPM module to update Windows Registry](building-for-cross-platform.md#Reading and Writing to the Windows Registry in-process from Node.js)
```
Comment (mousetraps) [inline README.md]: How about we label this `Windows services and settings` instead? That way we can expand beyond the registry.

File: building-for-cross-platform.md [hunk 2]
```diff
@@ -1,3 +1,119 @@
 ## Writing cross-platform modules and apps
 * http://shapeshed.com/writing-cross-platform-node/
 * https://gist.github.com/domenic/2790533
+
+### Reading and Writing to the Windows Registry in-process from Node.js
```
Comment (mousetraps) [inline building-for-cross-platform.md]: Let's move this into a page of it's own with the aforementioned title, and then make this a subheading of that.

**Comment by mousetraps** [comment]
@ritazh just left a couple comments re: refactoring and :+1: after that!

**Comment by ritazh** [comment]
Done! Made all the suggested changes. :smile:

**Comment by mousetraps** [comment]
Awesome, thanks for the contribution!

**Pull request merged**